src/appdistribution/options-parser-util.ts

```diff
@@ -62,3 +64,105 @@ export function getAppName(options: any): string {
   const appId = options.app;
   return `projects/${appId.split(":")[1]}/apps/${appId}`;
 }
+
+/**
+ * Takes in comma separated string or a path to a comma/new line separated file
+ * and converts the input into a string[] of test device strings. Value takes precedent
+ * over file.
+ */
+export function getTestDevices(value: string, file: string): TestDevice[] {
+  // If there is no value then the file gets parsed into a string to be split
+  if (!value && file) {
+    ensureFileExists(file);
+    value = fs.readFileSync(file, "utf8");
+  }
+
+  if (!value) {
+    return [];
+  }
+
+  // The value is split into a string[]
+  const deviceStrings = value
+    .split(/[;\n]/)
+    .map((entry) => entry.trim())
+    .filter((entry) => !!entry);
+
+  return deviceStrings.map((str) => parseTestDevice(str));
```

Comment:
Nit: Since you're already chaining here, you could go a step further and: 
```
return value
    .split(/[;\n]/)
    .map((entry) => entry.trim())
    .filter((entry) => !!entry)
    .map((str) => parseTestDevice(str));
```

Reply:
Good call, done